aws-s3/spring-cloud-starter-stream-source-s3/src/main/java/org/springframework/cloud/stream/app/s3/source/AmazonS3SourceConfiguration.java

```diff
@@ -55,6 +57,11 @@
 	private ResourceIdResolver resourceIdResolver;
 
 	@Bean
+	public TriggerProperties triggerProperties() {
+		return new TriggerPropertiesMaxMessagesDefaultUnlimited();
```

Comment:
The bigger concern isn't how it is wired, but the fact we need a TriggerPropertiesMaxMessagesDefaultUnlimited class at all.  It is all around getting good metadata files generated.  We could use an `@EnableConfigurationProperties(TriggerPropertiesMaxMessagesDefaultUnlimited.class) here, but it is really just the same thing.  I believe I went this route probably out of evolution in terms of trying to use the same class and doing some extra configuration in the `@Bean` method vs. just newing it up.